Mathlib/Analysis/NormedSpace/HahnBanach/Separation.lean

```diff
@@ -200,3 +202,103 @@ theorem iInter_halfspaces_eq (hs₁ : Convex ℝ s) (hs₂ : IsClosed s) :
   obtain ⟨l, s, hlA, hl⟩ := geometric_hahn_banach_closed_point hs₁ hs₂ h
   obtain ⟨y, hy, hxy⟩ := hx l
   exact ((hxy.trans_lt (hlA y hy)).trans hl).not_le le_rfl
+
+section RCLike
+
+open RCLike
+
+variable [RCLike 𝕜] [TopologicalSpace E] [AddCommGroup E] [TopologicalAddGroup E]
+  [Module 𝕜 E] [Module ℝ E] [ContinuousSMul 𝕜 E] [IsScalarTower ℝ 𝕜 E]
+
+/--Real linear extension of continuous extension of `LinearMap.extendTo𝕜'` -/
+noncomputable def LinTo𝕜' : (E →L[ℝ] ℝ) →ₗ[ℝ] (E →L[𝕜] 𝕜) :=
+  letI to𝕜 (fr : (E →L[ℝ] ℝ)) : (E →L[𝕜] 𝕜) :=
+    { toLinearMap := LinearMap.extendTo𝕜' fr
+      cont := show Continuous fun x ↦ (fr x : 𝕜) - (I : 𝕜) * (fr ((I : 𝕜) • x) : 𝕜) by fun_prop }
+  have h fr x : to𝕜 fr x = ((fr x : 𝕜) - (I : 𝕜) * (fr ((I : 𝕜) • x) : 𝕜)) := rfl
+  { toFun := to𝕜
+    map_add' := by intros; ext; simp [h]; ring
+    map_smul' := by intros; ext; simp [h, real_smul_eq_coe_mul]; ring }
+
+@[simp]
+lemma real_of_LinTo𝕜' (g : E →L[ℝ] ℝ) (x : E) :  re ((LinTo𝕜' g) x : 𝕜) = g x := by
+  have h g (x : E) : LinTo𝕜' g x = ((g x : 𝕜) - (I : 𝕜) * (g ((I : 𝕜) • x) : 𝕜)) := rfl
+  simp only [h , map_sub, ofReal_re, mul_re, I_re, zero_mul, ofReal_im, mul_zero,
+  sub_self, sub_zero]
+
+variable [ContinuousSMul ℝ E]
+
+theorem separate_convex_open_set_RCLike  {s : Set E}
+    (hs₀ : (0 : E) ∈ s) (hs₁ : Convex ℝ s) (hs₂ : IsOpen s) {x₀ : E} (hx₀ : x₀ ∉ s) :
+    ∃ f : E →L[𝕜] 𝕜, re (f x₀) = 1 ∧ ∀ x ∈ s, re (f x) < 1 := by
+  obtain ⟨g, hg⟩ := separate_convex_open_set hs₀ hs₁ hs₂ hx₀; use LinTo𝕜' g
+  simp only [real_of_LinTo𝕜']; exact hg
+
+theorem geometric_hahn_banach_open_RCLike (hs₁ : Convex ℝ s) (hs₂ : IsOpen s) (ht : Convex ℝ t)
```

Comment:
Maybe these should go into the RCLike namespace

Reply:
If they're in the `RCLike` namespace, then there is no reason to append `RCLike` to the names of the lemmas.

Reply:
Interesting. What is the better style move, here? Should I just remove the `RCLike` from the names of the lemmas and put these (named identically) lemmas, or should I keep them out of the namespace and include the underscored qualification? (I imagine this is sorted by the naming convention somehow, but I'm still internalizing this...)

Reply:
By the way, I tried removing the `_RCLike` from these names, but Lean then is still identifying the results with those named identically (aside from the `RCLike.` appearing above.) I get structural recursion errors.

Reply:
Those errors come from the fact that what you invoke (say) `geometric_hahn_banach_open` from `RCLike.geometric_hahn_banach_open`, it looks like a recursive call. To make sure that Lean interprets it as invoking the original theorem, you can write `_root_.geometric_hahn_banach_open` to specify you want the one from the root namespace.

As for whether we want the lemmas in the `RCLike` namespace or with `rclike` appended to the names (it should probably be spelled like that in this case), I don't really have a strong preference.